src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,30 @@
 import React from 'react';
+import { useAppSelector } from '../../app/store';
+import { actions, filterSelector } from '../../features/filter';
+import { useDispatch } from 'react-redux';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const filter = useAppSelector(filterSelector);
+  const dispatch = useDispatch();
+
   return (
     <form
-      className="field has-addons"
       onSubmit={event => event.preventDefault()}
+      className="field has-addons"
     >
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
-            <option value="all">All</option>
-            <option value="active">Active</option>
-            <option value="completed">Completed</option>
+          <select
+            value={filter.status}
+            onChange={e =>
+              dispatch(actions.setStatusFilter(e.target.value as Status))
```

Comment:
Ensure that `filter.status` is always defined to prevent potential runtime errors. Verify that the `setStatusFilter` action correctly handles the `Status` type.